api/specs.proto

```diff
@@ -72,19 +72,7 @@ message ServiceSpec {
 	// UpdateConfig controls the rate and policy of updates.
 	UpdateConfig update = 6;
 
-	// NetworkAttachmentConfig specifies how a service should be attached to a particular network.
-	//
-	// For now, this is a simple struct, but this can include future information
-	// instructing Swarm on how this service should work on the particular
-	// network.
-	message NetworkAttachmentConfig {
-		// Target specifies the target network for attachment. This value may be a
-		// network name or identifier. Only identifiers are supported at this time.
-		string target = 1;
-		// Aliases specifies a list of discoverable alternate names for the service on this Target.
-		repeated string aliases = 2;
-	}
-	repeated NetworkAttachmentConfig networks = 7;
+	repeated NetworkAttachmentConfig networks = 7 [deprecated=true];
```

Comment:
The backward compatibility from API point of view will be addressed in engine-api. It will accept NetworkAttachmentConfig in both ServiceSpec and TaskSpec but will always add them to TaskSpec in controlapi.

The backward compatibility from state migration point of view is not being done as part of this PR. The design for this roughly is to do inline migration in allocator when we upgrade to 1.13 when we need to upgrade services created from old version to new version to update migrate the `NetworkAttachmentConfig` from ServiceSpec to TaskSpec. I just did not want to tackle that as well as part of this PR to reduce the scope of this PR. If it helps we can create a P1 issue to make sure we don't lose track of that.